### ISSUE TITLE
Does Not Reliably Detect Origin / Remotes

### ISSUE THREAD

**Issue by GuyPaddock** [CircleCI-Public/circleci-config-generator#1]
The current script does not properly handle repositories that have:
- Multiple remotes
- A remote that does not end its URL in `.git`.

For example. here's some output from `git remote -v` that is not parsed properly:
```
matt    git@github.com:matt/some-project (fetch)
matt    git@github.com:matt/some-project (push)
origin  git@github.com:GuyPaddock/some-project.git (fetch)
origin  git@github.com:GuyPaddock/some-project.git (push)
production      git@github.com:org/some-project.git (fetch)
production      git@github.com:org/some-project.git (push)
```

This leads to the following error:
```
fatal: remote error:
   is not a valid repository name
  Email support@github.com for help
```

It looks like the following line:
```
project="$(git remote -v | awk -F'[@:.]' 'NR==1{ print $4 }')"
```

Should be changed to:
```
project="$(git remote -v | grep origin | sed 's/\t/ /' | cut -d' ' -f2 | sort | uniq | awk -F'[@:.]' 'NR==1{ print $4 }')"
```

This ensures:
- The origin is used instead of the first repo that appears.
- The delimiters between fields is always a space (instead of tabs between most fields and a space between the remote URL and the `(fetch)` or `(push)` indicator.

**Comment by eddiewebb** [comment]
Would you be willing to submit this as PR @GuyPaddock ?

**Comment by woodardj** [comment]
Would be nice to get this fixed.

**Comment by eddiewebb** [comment]
@GuyPaddock, @woodardj , please try the referenced changes in https://github.com/CircleCI-Public/circleci-config-generator/blob/issue-5-issue-1/cci-config-generator.sh and confirm this works for multiple remotes (will always use origin).

**Comment by woodardj** [comment]
Negative.

Here's my remote -v list:
```
Rockhopper:<<project-name>> woodardj[master]$ git remote -v
heroku	https://git.heroku.com/<<project-name>>.git (fetch)
heroku	https://git.heroku.com/<<project-name>>.git (push)
origin	git@github.com:woodardj/<<project-name>>.git (fetch)
origin	git@github.com:woodardj/<<project-name>>.git (push)
```

Here's what I see when I run the new script:
```
Rockhopper:<<project-name>> woodardj[master]$ ./cci-config-generator.sh 
Gathering info please wait...

 !  Invalid path.
 !  Syntax is: git@heroku.com:<app>.git where <app> is your app's name.
 !  
 !  SSH Key Fingerprint: <<snip>>

fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
git origin references `` hosted by heroku, which will be used as source.
Paste your CircleCI API token here: ^C
```

**Comment by woodardj** [comment]
`woodardj/<<project-name>>` 👍

**Comment by eddiewebb** [comment]
Hi @woodardj  thanks for validating.  Your comment above changed, so I just want to be clear.  THe latest script (on the branch above)  should now support multiple remotes, always choosing origin.  The relevant portions are:

```
vcs_provider="$(git remote get-url --push origin | perl -ne 'print $1 if /([A-Za-z]*)\.(com|org)/')"
project="$(git remote get-url --push origin | perl -ne 'print $1 if /([^\/:]*\/[^\/]*?)(.git)?$/')"
```

So if the script doesn't run, please execute those and then grab their values (i.e. `echo "$project  at $vcs_provider"`)


I'm testing locally against various patterns noted in all 3 open issues, want to be sure I haven't omitted something!

**Comment by woodardj** [comment]
Looks like it's all good now — this script was super smooth and easy once it worked! Thanks for putting it together! Now I've just gotta figure out why one of my repos is failing.

**Comment by eddiewebb** [comment]
Fixed in #9
